packages/pwa-kit-dev/src/ssr/server/public-path.js

```diff
@@ -9,12 +9,24 @@
  * This file is used to dynamically set the webpack public path used by HMR via the global webpack variable
  * __webpack_public_path__
  * See https://webpack.js.org/guides/public-path/
- *
- * Previously, we hard coded the public path in our webpack config to '/mobify/bundle/development/'
- * but we need something more dynamic to support namespaced /mobify paths.
  */
 
-import {bundleBasePath} from '@salesforce/pwa-kit-runtime/utils/ssr-namespace-paths'
-
 /* global __webpack_public_path__: writable */
-__webpack_public_path__ = `${bundleBasePath}/development/`
+
+/**
+ * TODO - __webpack_public_path__ must be the first thing set on client side local environments
+ * for HMR to work!
+ * However, when this value needs to be set, window.__config__ is not yet hydrated so we
+ * do not have access to the base path
+ *
+ * Resolve this to allow the local development server to use namespaces.
+ *
+ * import {bundleBasePath} from '@salesforce/pwa-kit-runtime/utils/ssr-namespace-paths'
+ * const getPublicPath = () => `${bundleBasePath()}/development/`
+ * console.log(getPublicPath())
+ *
+ * __webpack_public_path__ = getPublicPath()
+ */
+
+// Setting this so that *.hot-update.json requests are resolving
+__webpack_public_path__ = '/mobify/bundle/development'
```

Comment:
Not being able to set a base path here is a blocker for getting base paths to work on a local dev server (unless you have some way to hard code this). 

Supporting base paths on the local dev server is not a critical requirement at this time so I can either leave this as-is or delete this from the PR depending on what the team thinks.

Reply:
I think one potential approach is to resolve `public-path.js` logic before the App entry point.

https://github.com/SalesforceCommerceCloud/pwa-kit/blob/b79d8d1bda6e5175ce640d6f012e951d19460121/packages/pwa-kit-dev/src/configs/webpack/config.js#L437-L439

```
entry: {
                    main: [
                        // Adding the public-path.js module as the first entry
                        path.resolve(__dirname, 'src', 'public-path.js'),
                        getAppEntryPoint()
                    ]
                },
```

Reply:
Thanks for the suggestion @adamraya ! 

I've been mulling over the public-path on local environments bit and just pushed a solution that works.
The new change is to set the public path from within `build-dev-server.js` if HMR is enabled before `webpackHotMiddleware` is invoked. Following this change, I'll be deleting `public-path.js`